src/includes/Services/CollectCmsDataService.php

```diff
@@ -0,0 +1,222 @@
+		$payload_formatter = null,
+		$security_helper = null,
+		$option_proxy = null,
+		$theme_proxy = null,
+		$functions_proxy = null
+	) {
+		$this->alma_settings     = isset( $alma_settings ) ? $alma_settings : new AlmaSettings();
+		$this->alma_logger       = isset( $alma_logger ) ? $alma_logger : new AlmaLogger();
+		$this->payload_formatter = isset( $payload_formatter ) ? $payload_formatter : new PayloadFormatter();
+		$this->security_helper   = isset( $security_helper ) ? $security_helper : ( new SecurityHelperBuilder() )->get_instance();
+		$this->option_proxy      = isset( $option_proxy ) ? $option_proxy : new OptionProxy();
+		$this->theme_proxy       = isset( $theme_proxy ) ? $theme_proxy : new ThemeProxy();
+		$this->functions_proxy   = isset( $functions_proxy ) ? $functions_proxy : new FunctionsProxy();
+	}
+
+	/**
+	 * Handle collect cms data
+	 *
+	 * @return void
+	 */
+	public function handle_collect_cms_data() {
+		if ( ! array_key_exists( 'HTTP_X_ALMA_SIGNATURE', $_SERVER ) ) {
+			$this->alma_logger->error( 'Header key X-Alma-Signature doesn\'t exist' );
+			$this->functions_proxy->send_http_response( array( 'error' => 'Header key X-Alma-Signature doesn\'t exist' ), 403 );
+			return;
+		}
+
+		try {
+			$this->security_helper->validate_collect_data_signature(
+				$this->alma_settings->get_active_merchant_id(),
+				$this->alma_settings->get_active_api_key(),
+				$_SERVER['HTTP_X_ALMA_SIGNATURE']
+			);
+			$this->functions_proxy->send_http_response( $this->payload_formatter->formatConfigurationPayload( $this->get_cms_info(), $this->get_cms_features() ), 200 );
+		} catch ( AlmaInvalidSignatureException $e ) {
+			$this->alma_logger->error( $e->getMessage() );
+			$this->functions_proxy->send_http_response( array( 'error' => $e->getMessage() ), 403 );
+		}
+	}
+
+	/**
+	 * Get CMS features.
+	 *
+	 * @return CmsFeatures
+	 */
+	private function get_cms_features() {
+		return new CmsFeatures(
+			array(
+				'alma_enabled'             => $this->alma_settings->is_enabled(),
+				'widget_cart_activated'    => 'yes' === $this->alma_settings->display_cart_eligibility,
+				'widget_product_activated' => 'yes' === $this->alma_settings->display_product_eligibility,
+				'used_fee_plans'           => $this->format_fee_plans(),
+				'in_page_activated'        => $this->alma_settings->display_in_page,
+				'log_activated'            => $this->alma_settings->debug,
+				'excluded_categories'      => $this->alma_settings->excluded_products_list,
+				'is_multisite'             => is_multisite(),
+			)
+		);
+	}
+
+	/**
+	 * Get CMS info.
+	 *
+	 * @return CmsInfo
+	 */
+	private function get_cms_info() {
+		$active_plugins = $this->option_proxy->get_option( 'active_plugins' );
+
+		return new CmsInfo(
+			array(
+				'cms_name'              => 'WooCommerce',
+				'cms_version'           => $this->option_proxy->get_option( 'woocommerce_version' ),
+				'third_parties_plugins' => $this->format_third_party_modules( $active_plugins ),
```

Comment:
it's a private function not reusable, you can add the get option code in the method and remove injection

Reply:
Done